### PR TITLE
Fixes and improvements

### PR DESCRIPTION
**Pull request by dennisreimann**
Multiple fixes for the tests as well as the payment and list channel methods in LND and Eclair. Those got surfaced by the recent flakyness in the tests.

### CODE REVIEW

File: src/BTCPayServer.Lightning.All/ConnectChannels.cs [hunk 1]
```diff
@@ -50,8 +50,9 @@ public static async Task ConnectAll(RPCClient cashCow, IEnumerable<ILightningCli
         }
         public static async Task CreateChannel(RPCClient cashCow, ILightningClient sender, ILightningClient dest)
         {
+            var amount = new LightMoney(1234); // use arbitrary amount to check if channel exists
```
Comment (dennisreimann) [inline src/BTCPayServer.Lightning.All/ConnectChannels.cs:53]: Using this arbitrary amount to distinguish it from the value used in `CanPayInvoiceAndReceive`.

File: src/BTCPayServer.Lightning.Eclair/EclairLightningClient.cs [hunk 2]
```diff
@@ -134,12 +134,12 @@ public async Task<PayResponse> Pay(string bolt11, PayInvoiceParams payParams, Ca
...
-                var req = new PayInvoiceRequest()
+                var req = new PayInvoiceRequest
                 {
-                    Invoice = bolt11
+                    Invoice = bolt11,
+                    MaxFeePct = payParams?.MaxFeePercent != null ? (int)Math.Round(payParams.MaxFeePercent.Value) : (int?)null,
+                    MaxFeeFlatSat = payParams?.MaxFeeFlat?.Satoshi
                 };
-                req.MaxFeePct = payParams?.MaxFeePercent;
-                req.MaxFeeFlatSat = payParams?.MaxFeeFlat?.Satoshi;
```
Comment (dennisreimann) [inline src/BTCPayServer.Lightning.Eclair/EclairLightningClient.cs:142]: Only a cleanup.

File: src/BTCPayServer.Lightning.Eclair/EclairLightningClient.cs [hunk 3]
```diff
@@ -255,13 +256,13 @@ public async Task<LightningChannel[]> ListChannels(CancellationToken cancellatio
...
 
-                return new LightningChannel()
+                return new LightningChannel
                 {
                     IsPublic = response.Data.Commitments.IsPublic,
                     RemoteNode = new PubKey(response.NodeId),
                     IsActive = response.State == "NORMAL",
                     LocalBalance = new LightMoney(response.Data.Commitments.LocalCommit.Spec.ToLocalMsat),
-                    Capacity = new LightMoney(response.Data.Commitments.CommitInput.AmountSatoshis),
+                    Capacity = new LightMoney(response.Data.Commitments.CommitInput.AmountSatoshis, LightMoneyUnit.Satoshi),
```
Comment (dennisreimann) [inline src/BTCPayServer.Lightning.Eclair/EclairLightningClient.cs:265]: The amount got interpreted as millisats before.

File: src/BTCPayServer.Lightning.Eclair/Models/ChannelResponse.cs [hunk 4]
```diff
@@ -82,8 +82,11 @@ public partial class NextRemoteCommit
         public partial class Spec
         {
             public List<object> Htlcs { get; set; }
+            [JsonProperty("commitTxFeerate")]
             public long FeeratePerKw { get; set; }
+            [JsonProperty("toLocal")]
             public ulong ToLocalMsat { get; set; }
+            [JsonProperty("toRemote")]
```
Comment (dennisreimann) [inline src/BTCPayServer.Lightning.Eclair/Models/ChannelResponse.cs:89]: Maybe those changed with an Eclair update.

File: src/BTCPayServer.Lightning.Eclair/Models/PayInvoiceRequest.cs [hunk 5]
```diff
@@ -5,7 +5,7 @@ public class PayInvoiceRequest
         public string Invoice { get; set; }
         public int? AmountMsat { get; set; }
         public int? MaxAttempts { get; set; }
-        public double? MaxFeePct { get; set; }
+        public int? MaxFeePct { get; set; }
```
Comment (dennisreimann) [inline src/BTCPayServer.Lightning.Eclair/Models/PayInvoiceRequest.cs:8]: Needs to be `int`, [reference](https://acinq.github.io/eclair/#payinvoice).

File: src/BTCPayServer.Lightning.LND/LndClient.cs [hunk 6]
```diff
@@ -219,19 +219,19 @@ await this.SwaggerClient.CancelInvoiceAsync(new InvoicesrpcCancelInvoiceMsg()
...
+                    select new LightningChannel {
                         RemoteNode = new PubKey(c.Remote_pubkey),
-                        IsPublic = !(c.Private ?? false) ,
-                        IsActive = (c.Active == null ? false : c.Active.Value),
-                        Capacity = c.Capacity,
-                        LocalBalance = c.Local_balance,
+                        IsPublic = !(c.Private ?? false),
+                        IsActive = c.Active ?? false,
+                        Capacity = LightMoney.Satoshis(long.Parse(c.Capacity)),
+                        LocalBalance = LightMoney.Satoshis(long.Parse(c.Local_balance)),
```
Comment (dennisreimann) [inline src/BTCPayServer.Lightning.LND/LndClient.cs:234]: Those are strings and need to get parsed, otherwise `LightMoney` assumes BTC values.

**Review by Kukks — APPROVED** [review]

**Pull request merged**